src/rhsmlib/facts/dmidecodeparser.py

```diff
@@ -0,0 +1,209 @@
+            )
+            self._parse_lines(proc.stdout)
+        except subprocess.SubprocessError:
+            error = proc.stderr.read()
+            log.warning(f"Error with dmidecode subprocess: {error}")
+
+    def parse_file(self, filename):
+        with open(filename, "r") as f:
+            self._parse_lines(f)
+
+    def _parse_lines(self, fd):
+        class ParsingState(enum.Enum):
+            NONE = enum.auto()
+            IN_SECTION = enum.auto()
+            IN_RECORD = enum.auto()
+            IN_BLOCK = enum.auto()
+
+        state = ParsingState.NONE
+        current_handle = None
+        current_key = None
+        current_dmi_type = None
+        re_handle = re.compile("^Handle\\s+(.+),\\s+DMI\\s+type\\s+(\\d+),\\s+(\\d+)\\s+bytes$")
+
+        while True:
+            line = fd.readline()
+            if not line:
+                break
+
+            line = line.rstrip()
+            if len(line) == 0:
+                state = ParsingState.NONE
+                current_handle = None
+                current_key = None
+                current_dmi_type = None
+                continue
+
+            if line.startswith("Handle "):
+                m = re_handle.fullmatch(line)
+                if m:
+                    state = ParsingState.IN_SECTION
+                    current_handle = int(m[1], base=16)
+                    current_dmi_type = int(m[2])
+                    self._data[current_handle] = dict()
+                    handles = self._dmi_types[current_dmi_type].get("handles", [])
+                    handles.append(current_handle)
+                    self._dmi_types[current_dmi_type]["handles"] = handles
+                    continue
+
+            if state == ParsingState.IN_SECTION:
+                # section name
+                self._dmi_types[current_dmi_type]["name"] = line
+                state = ParsingState.IN_RECORD
+                continue
+
+            if state == ParsingState.IN_RECORD or state == ParsingState.IN_BLOCK:
+                if line.startswith("\t\t"):
+                    if state == ParsingState.IN_RECORD:
+                        # had a value, drop it
+                        with contextlib.suppress(KeyError):
+                            del self._data[current_handle][current_key]
+                    value = line[2:]
+                    try:
+                        current_value = self._data[current_handle][current_key]
+                        if not isinstance(current_value, list):
+                            current_value = [current_value]
+                        current_value.append(value)
+                        self._data[current_handle][current_key] = current_value
+                    except KeyError:
+                        self._data[current_handle][current_key] = value
+                    state = ParsingState.IN_BLOCK
+                elif line.startswith("\t"):
+                    parts = line[1:].split(":", maxsplit=1)
```

Comment:
hm since there is only a leading `\t` in the line (and not two, that is handled before this block), then we know the only thing that needs to be removed is that only `\t`

Reply:
OK, I'm only afraid that they will add spaces and not tabs in the future for something.